app/models/foreman_salt/concerns/host_managed_extensions.rb

```diff
@@ -97,28 +104,58 @@ def salt_modules_in_host_environment
 
       private
 
-      def accept_salt_key
+      def ensure_salt_autosign
+        remove_salt_autosign
+        create_salt_autosign
+      end
+
+      def remove_salt_autosign
+        key = self.salt_autosign_key
+        unless key.nil?
+          Rails.logger.info("Remove salt autosign key for host #{fqdn}")
+          begin
+            api = ProxyAPI::Salt.new(:url => salt_proxy.url)
+            api.autosign_remove_key(key)
+          rescue Foreman::Exception => e
+            Rails.logger.warn("Unable to remove salt autosign for #{fqdn}: #{e}")
+          end
+        end
+      end
+
+      def generate_provisioning_key
+          SecureRandom.hex(10)
+      end
+
+      def create_salt_autosign
         begin
-          Rails.logger.info("Host #{fqdn} is built, accepting Salt key")
-          key = ForemanSalt::SmartProxies::SaltKeys.find(salt_proxy, fqdn)
-          key&.accept
+          Rails.logger.info("Create salt autosign key for host #{fqdn}")
+          api = ProxyAPI::Salt.new(:url => salt_proxy.url)
+          key = generate_provisioning_key
+          api.autosign_create_key(key)
+          update(:salt_autosign_key => key)
+          update(:salt_status => ForemanSalt::SaltStatus.minion_auth_waiting)
         rescue Foreman::Exception => e
-          Rails.logger.warn("Unable to accept key for #{fqdn}: #{e}")
+          Rails.logger.warn("Unable to create salt autosign for #{fqdn}: #{e}")
         end
       end
 
-      def delete_salt_key
+      def derive_salt_grains(use_autosign: False)
```

Comment:
Works fine like [this](https://www.rubyguides.com/2018/06/rubys-method-arguments/). Since the position is not important at this point, one could also use `use_autosign = false`.

Reply:
ok. Fine with it. Thanks!

Reply:
should this have been `false` and not `False`?

Reply:
let's see in https://github.com/theforeman/foreman_salt/pull/160

Reply:
I wonder if this method is used anywhere except in tests.

Reply:
https://github.com/theforeman/foreman/pull/8489 is trying to use it in a template

Reply:
By the way, from my experience, if you follow rubocop guidelines, you would either not set a default value for Booleans or use `use_autosign: false`. The latter one forces you to explicitly write out the parameter name which is considered more readable code compared to simply passing boolean parameters like `true` or `false`.